AgentDiscoveries-Frontend/app/src/components/profile.js

```diff
@@ -84,7 +84,7 @@ export default class Profile extends React.Component {
         var userId = window.localStorage.getItem("UserId");
         readAPI("/v1/api/pictures", userId).then(response => response.json())
```

Comment:
I'm still slightly suspicious of all this conversion code... I assume you've tested this?

- I think it should no longer be returning json, so `response.json()` seems suspect?
- Transforming binary data via a string (even if you convert it back later) is a sure way to get corruption.

Have a look at the example at https://developer.mozilla.org/en-US/docs/Web/API/Body/blob - I think this is exactly what you want!

*Warning*: this API is experimental and not supported in IE. Do you know if we've already ditched IE support, or do we need to polyfill?

Reply:
I had tested and it seemed to work fine. I've had another go at just using blobs like you mentioned above, and after a while of getting nowhere realised that the responseTransformer that we were using was ensuring that the response was still a json, hence why I had to leave the json conversion above, and this meant blob conversion wasn't working as intended. Should now be nicer, and still functional!